tools/common/Application.cs

```diff
@@ -108,6 +110,119 @@ public partial class Application
 
 		public BuildTarget BuildTarget;
 
+		public bool? DisableLldbAttach = null; // Only applicable to Xamarin.Mac
+		public bool? DisableOmitFramePointer = null; // Only applicable to Xamarin.Mac
+		public string CustomBundleName = "MonoBundle"; // Only applicable to Xamarin.Mac
+
+		public bool? UseMonoFramework;
+
+		// The bitcode mode to compile to.
+		// This variable does not apply to macOS, because there's no bitcode on macOS.
+		public BitCodeMode BitCodeMode { get; set; }
+
+		public bool EnableAsmOnlyBitCode { get { return BitCodeMode == BitCodeMode.ASMOnly; } }
+		public bool EnableLLVMOnlyBitCode { get { return BitCodeMode == BitCodeMode.LLVMOnly; } }
+		public bool EnableMarkerOnlyBitCode { get { return BitCodeMode == BitCodeMode.MarkerOnly; } }
+		public bool EnableBitCode { get { return BitCodeMode != BitCodeMode.None; } }
+
+		// assembly_build_targets describes what kind of native code each assembly should be compiled into for mobile targets (iOS, tvOS, watchOS).
+		// An assembly can be compiled into: static object (.o), dynamic library (.dylib) or a framework (.framework).
+		// In the case of a framework, each framework may contain the native code for multiple assemblies.
+		// This variable does not apply to macOS (if assemblies are AOT-compiled, the AOT compiler will output a .dylib next to the assembly and there's nothing extra for us)
+		Dictionary<string, Tuple<AssemblyBuildTarget, string>> assembly_build_targets = new Dictionary<string, Tuple<AssemblyBuildTarget, string>> ();
+
+		// How Mono should be embedded into the app.
+		public AssemblyBuildTarget LibMonoLinkMode {
+			get {
+				if (Platform == ApplePlatform.MacOSX) {
+					// This property was implemented for iOS, but might be re-used for macOS if desired after testing to verify it works as expected.
+					throw ErrorHelper.CreateError (99, Errors.MX0099, "LibMonoLinkMode isn't a valid operation for macOS apps.");
+				}
+
+				if (Embeddinator) {
+					return AssemblyBuildTarget.StaticObject;
+				} else if (HasFrameworks || UseMonoFramework.Value) {
+					return AssemblyBuildTarget.Framework;
+				} else if (HasDynamicLibraries) {
+					return AssemblyBuildTarget.DynamicLibrary;
+				} else {
+					return AssemblyBuildTarget.StaticObject;
+				}
```

Comment:
We have this if also in LibXamarinLinkMode, I think we could share it so that we do not have mismatching results in the future.

Reply:
The implementations for LibMonoLinkMode and LibXamarinLinkMode aren't identical.